File: app/src/main/java/com/apt/delegate/source/IRemoteDataSource.java
package com.apt.delegate.source;

import com.annotation.Delegate;
import com.annotation.SingleDelegate;

@SingleDelegate(
        classNameImpl = "RemoteDataSourceImpl",
        delegate = @Delegate(
                delegatePackage = "com.apt.delegate.source.delegate",
                delegateClassName = "RemoteDelegate",
                delegateSimpleName = "remoteDelegate"
        ))
public interface IRemoteDataSource extends IOtherDataSource {

    void getRemoteUser();

    void getRemoteUserList();

    void getRemoteBook();
}

File: app/src/main/java/com/apt/delegate/source/IRemoteDataSource.kt
package com.apt.delegate.source

import com.annotation.Delegate
import com.annotation.SingleDelegate

@SingleDelegate(
        classNameImpl = "RemoteDataSourceImpl",
        delegate = Delegate(
                delegatePackage = "com.apt.delegate.source.delegate",
                delegateClassName = "RemoteDelegate",
                delegateSimpleName = "remoteDelegate"))
interface IRemoteDataSource : IOtherDataSource {

    fun getRemoteUser()

    fun getRemoteUserList()

    fun getRemoteBook()
}
